File: backend/NailApi/NailApi/Models/Product.cs
using System.ComponentModel.DataAnnotations;

namespace NailApi.Models
{
    public class Product
    {
        [Key]
        public int ProductId { get; set; }
        public string ProductSku { get; set; } = string.Empty;
        public string ProductName { get; set; } = string.Empty;
        public decimal ProductPrice { get; set; }
        public string ProductShortName { get; set; } = string.Empty;
        public string ProductDescription { get; set; } = string.Empty;
        public DateTime CreatedDate { get; set; }
        public string DeliveryTimeSpan { get; set; } = string.Empty;
        public int CategoryId { get; set; }
        public string ProductImageUrl { get; set; } = string.Empty;
        public bool IsActive { get; set; } = true;


    }

}

File: frontend-admin/src/app/models/product.ts
import { ProductImage } from "./ProductImage";

export interface Product {
  id?: number;
  productId:number;
    productSku: string;
    productName: string;
    productPrice: number;
    productShortName: string;
    productDescription: string;
    deliveryTimeSpan: string;
    categoryId: number;   
    categoryName:string;
    image?: File; 
    productImageUrl:string;
    isActive: boolean;
    additionalImages?: ProductImage[]; // Add this property
}